File: 2022/06/06.py
with open("inputs/6.txt") as file:
    input_chars = list(file.read().strip())

def get_unique_sequence_end_position(sequence_length):
    for i in range(len(input_chars)):
        if len(set(input_chars[i:i+sequence_length])) == sequence_length:
            return i + sequence_length

def part_1():
    return get_unique_sequence_end_position(4)

def part_2():
    return get_unique_sequence_end_position(14)


print(f"Part 1: {part_1()}")
print(f"Part 2: {part_2()}")

File: 2022/06/06.rb
# frozen_string_literal: true

path = File.expand_path('../inputs/6.txt', __dir__)

input = File.read(path, chomp: true)

first_four_char_unique_sequence = input.chars.each_cons(4).find do |current_characters|
  current_characters.uniq.length == 4
end

part1 = input.index(first_four_char_unique_sequence.join) + 4

first_fourteen_char_unique_sequence = input.chars.each_cons(14).find do |current_characters|
  current_characters.uniq.length == 14
end

part2 = input.index(first_fourteen_char_unique_sequence.join) + 14

puts "part 1:  #{part1} "
puts "part 2:  #{part2} "
